zpm.c

```diff
@@ -227,13 +249,28 @@ char* get_zpm_plugin_list() {
         strcpy(listing, "Nothing to show.");
     }
 
+    free(plugin_list_path);
+
     return listing;
 }
 
 int plugins_update_local_clone() {
-    // iterate over plugins_list
-    // exec git pull --git-dir=
-    return 0;
+    int ret;
+    char* command = malloc(PATH_MAX);
+    char* listing = get_zpm_plugin_list();
+    char* plugin_name = strtok(listing, "\n");
+
+    while(plugin_name) {
+      strcpy(command, "cd ~/.zpm/plugins/");
+      strcat(command, plugin_name);
+      strcat(command, "; git pull");
+      ret = system(command);
+      plugin_name = strtok(NULL, "\n");
+    }
+    printf("Update plugins ...\n");
```

Comment:
This `printf` should be before the `while` loop, not after it. Be sure it reads "Updating plugins...".